### PR TITLE
Windows builds

### PR DESCRIPTION
**Pull request by teqdruid**
Unfortunately, I have to care about Windows as that is (internally) what our builds are based on. This PR just ensures that CIRCT builds on Windows.

- Adds a nightly job which does a Windows build.
- Fixes issues which cause the build to fail locally.
- Disables `llhd-sim` on Windows. (#375).
- Runs 'Release' build. I was unable (due to cmake issues, presumably in MLIR) to compile MLIR in `Release` and CIRCT in `Debug`. Compiling MLIR in `Debug` on Windows uses more space (~30 GB IIRC) than the runner has allocated to it (7 GB).

### CODE REVIEW

**Comment by teqdruid** [comment]
I finally got this working! Test build: https://github.com/teqdruid/circt/runs/1675536515

**Comment by lattner** [comment]
Awesome John!

**Comment by teqdruid** [comment]
I get the sense that nobody here feels qualified to review this particular PR (as per usual with cmake changes), so I'm just gonna merge it once the PR sanity build finishes.

**Comment by stephenneuendorffer** [comment]
I think you'll get something subtly broken if you compile MLIR and LLVM with different Debug or Release types, because MLIR depends on core parts of LLVM that deal with NDEBUG.

File: tools/llhd-sim/CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,11 @@ set(LIBS
         CIRCTLLHDSimEngine
         )
 
-add_llvm_executable(llhd-sim
- llhd-sim.cpp)
- 
-llvm_update_compile_flags(llhd-sim)
-target_link_libraries(llhd-sim PRIVATE ${LIBS})
+# llhd-sim fails to link on Windows with MSVC.
```
Comment (stephenneuendorffer) [inline tools/llhd-sim/CMakeLists.txt:12]: Why?

Comment (teqdruid) [reply tools/llhd-sim/CMakeLists.txt:12]: https://github.com/llvm/circt/issues/375

Comment (stephenneuendorffer) [reply tools/llhd-sim/CMakeLists.txt:12]: Oy

**Comment by stephenneuendorffer** [comment]
Thanks for doing this, BTW..  cmake is a thankless job 👍

**Pull request merged**